crates/fluvio/src/admin.rs

```diff
@@ -214,6 +214,38 @@ impl FluvioAdmin {
         Ok(())
     }
 
+    /// Forcibly delete object by key
+    /// key is dependent on spec, most are string but some allow multiple types.
+    ///
+    /// This method allows to delete objects marked as 'system'.
+    ///
+    /// For example, to delete a system topic:
+    ///
+    /// ```edition2021
+    /// use fluvio::Fluvio;
+    /// use fluvio::metadata::topic::TopicSpec;
+    ///
+    /// async fn delete_system_topic(name: String) -> anyhow::Result<()> {
+    ///     let fluvio = Fluvio::connect().await?;
+    ///     let admin = fluvio.admin().await;
+    ///     admin.force_delete::<TopicSpec>(name).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    #[instrument(skip(self, key))]
+    pub async fn force_delete<S>(&self, key: impl Into<S::DeleteKey>) -> Result<()>
```

Comment:
we should not try to create different variation of API for every new parameter.  Instead, let's try to remapped existing API as like this:
```
pub async fn delete<S>(&self, key: impl Into<S::DeleteRequest>)
```

And them add default mapping for string and so forth.

Reply:
But this is breaking change. It will come in a patch version bump,  users won't be happy

Reply:
This is not breaking change since old API should still work

Reply:
And it would be ok to bump up minor version if needed to be rather than create another API

Reply:
This won't work. `DeleteRequest` is generic over `<S: DeletableAdminSpec>` and to construct it you need to provide `S::DeleteKey`. Yes, mostly `DeleteKey=String` but not always. There is `CustomSpuKey` also. Anyway, the API needs the associated type, not concrete `String` or `CustomSpuKey`, I tried to do something like:
```rust
impl<S: DeletableAdminSpec> From<S::DeleteKey> for DeleteRequest<S> {
    fn from(value: S::DeleteKey) -> Self {
        todo!()
    }
}
```
or 
```rust
impl<T, S> From<T> for DeleteRequest<S>
where
    S: DeletableAdminSpec,
    T: Into<S::DeleteKey>,
{
    fn from(value: S::DeleteKey) -> Self {
        todo!()
    }
}
```
but it doesn't work because the `DeleteRequest` type also fits into `DeleteKey` trait bounds  and the compiler fails it with `conflicting trait impl` error.

Reply:
K.  Lets keep that as is and revisit this later when new WIT base is reformulated